### PR TITLE
Fixes to make the new functionality we've added opt-in

### LINKED ISSUE #5: Make tests pass again
The problem right now is that the latest changes done by @afinne are Postgres-only, since they use `UPSERT` etc, which is not available in SQLite.

Either we:

- make the code handle both scenarios, with if-statements or similar, _or_
- change Travis to use Postgres for its test runs.

### PR DESCRIPTION
**Pull request by perlun**
Namely, adds the following two new DSL keywords:

- ember_pouch_mode
- upsert_mode

These are pretty self-explanatory, and do what one would expect. They are _disabled_ by default, meaning we retain the `couch_tap` default behavior.

Since this makes the tests also work again, this PR closes #5.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,10 @@ The following example attempts to outline most of the key features of the DSL.
 # The couchdb database from which to request the changes feed
 changes "http://user:pass@host:port/invoicing" do
 
+  # upsert_mode       # Optional flag which can be enabled to take advantage of Postgres 9.5's support for INSERT CONFLICT,
+  #                   # e.g. upserts.
+  # ember_pouch_mode  # Optional flag which can be enabled if ember-pouch is being used to populate the CouchDB database.
```
Comment (perlun) [inline README.md]: Better name suggestion by @afinne - `relational_pouch_mode`. Will fix and re-test.

Comment (rallenecraft) [inline README.md]: Hade jag fel angående detta?

**Review by rallenecraft — APPROVED** [review]
Looks fine to me. But I'd prefer that @afinne approves this because he has studied couch-tap. I have not.

But from what I can see, all is fine. You decide if you wanna merge or get another approval from Andreas.

**Comment by perlun** [comment]
> But from what I can see, all is fine. You decide if you wanna merge or get another approval from Andreas.

Let's offload Andreas a bit. Merging away, thanks @rallenecraft!

**Pull request merged**